WIP: filterbar documentation (#537)
vivid-planet/comet: packages/admin-stories/src/docs/components/Table/FilterBar/FilterBar.stories.mdx

import { Meta, Story, Canvas } from "@storybook/addon-docs";

<Meta title="Docs/Components/Table/FilterBar/FilterBar" />

> # Filterbar

Reviewer: Filterbar or FilterBar? Why do you use both?

> The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include Final Form Fields.

Reviewer: ```suggestion
The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include Final Form fields.
```

import { Meta, Story, Canvas } from "@storybook/addon-docs";

> <Meta title="Docs/Components/Table/Filterbar/FilterBar" />

Reviewer: Filterbar or FilterBar? We should be consistent here. I'm rooting for FilterBar

Author: yes thats true :) i am also tending to FilterBar.

# Filterbar

> The Filterbar Component helps you create a horizontal orientated list of PopoverFields which can include all kind of sorts of Final Form Fields.

Reviewer: "all kind of sorts" is to casual for documentation

> The Intention of the Filterbar was to act as a Filter of a Table, which is also the main usecase for this Component.

Reviewer: ```suggestion
The intended use case for the FilterBar is to act as a filter on top of a table.
```

**The Filterbar only works inside of a Final Form.**

The Filterbar consists of four different components:

-   FilterBar
-   FilterBarPopoverFilter
-   FilterBarMoreFilters
-   FilterBarActiveFilterBadge

## Table Example

<Canvas>
    <Story id="stories-components-table-filterbar-filterbar-default--default" />
</Canvas>

## Advanced Informations

> Every FilterBarPopoverFilter Component contains its own FinalForm. So the Filterbar is kinda like a collection of final forms, which pass their values into a wrapping form.

Reviewer: This is an implementation detail. I don't think potential users need to know this

#### OnSubmit method of FilterBarPopoverFilter form

```
onSubmit={(values) => {
    for (const name in values) {
        outerForm.change(name, values[name]);
    }
}}
```

> The FilterBarActiveFilterBadge shows the current count of set values of every FilterBarPopoverFilter form. This componentas well as the calculation can be overritten.

Reviewer: ```suggestion
The FilterBarActiveFilterBadge shows the current count of set values of every FilterBarPopoverFilter form. This component as well as the calculation can be overridden.
```
